node/resolvers/search.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  Client,
+  Status,
+  PlaceAutocompleteType,
+  Language,
+  AddressType,
+  Place,
+  AddressGeometry,
+} from '@googlemaps/google-maps-services-js'
+import { pipe } from 'ramda'
+
+import countryRules from '../countries/rules'
+import { toAlpha3 } from '../countries/ISO'
+import { ISOAlpha2 } from '../countries/types'
+
+function revertRuleIndex(geolocationRules: GeolocationRules) {
+  return Object.entries(geolocationRules).reduce<{
+    [key: string]: GeolocationComponents
+  }>((acc, [propName, rule]) => {
+    for (let i = 0; i < rule!.types.length; i++) {
+      const type = rule!.types[i]
+      acc[type] = propName as GeolocationComponents
+    }
+    return acc
+  }, {})
+}
+
+function getCheckoutFieldName(
+  types: string[],
+  indexedRules: { [key: string]: GeolocationComponents }
+) {
+  const mappedType = types.find(type => indexedRules[type])
+  return mappedType ? indexedRules[mappedType] : null
+}
+
+// eslint-disable-next-line max-params
+function setAddressFieldValue(
+  address: Address,
+  fieldName: GeolocationComponents,
+  geolocationRules: GeolocationRules,
+  addressComponent: google.maps.GeocoderAddressComponent
+) {
+  const geolocationField = geolocationRules[fieldName]
+  address[fieldName] = addressComponent[geolocationField!.valueIn]
+  return address
+}
+
+function getCountry(place: Place) {
+  const countryComponent = place?.address_components?.find(
+    component => component.types.indexOf(AddressType.country) !== -1
+  )
+
+  return countryComponent
+    ? toAlpha3(countryComponent.short_name as ISOAlpha2)
+    : null
+}
+
+export const queries = {
+  getAddressByExternalId: async (
+    _: unknown,
+    args: { id: string },
+    ctx: Context
+  ): Promise<Address> => {
+    const { clients } = ctx
+    const { id } = args
+    const { apiKey } = await clients.apps.getAppSettings(
+      process.env.VTEX_APP_ID
+    )
+
+    const client = new Client({})
```

Comment:
this should be a client, much like the `GoogleGeolocation` client we have

Reply:
How to do that? 🤔

Reply:
(solved, just had to wrap the Google Client with a fake ExternalClient)